src/main/java/com/squareup/javapoet/AnnotatedTypeName.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (C) 2015 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.javapoet;
+
+import java.io.IOException;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public final class AnnotatedTypeName extends TypeName {
+
+  public static AnnotatedTypeName get(TypeName type, AnnotationSpec... annotations) {
+    return new AnnotatedTypeName(type, Arrays.asList(annotations));
+  }
+
+  public static AnnotatedTypeName get(Type type, Annotation... annotations) {
+    List<AnnotationSpec> specs = new ArrayList<>();
+    for (int i = 0; i < annotations.length; i++) {
```

Comment:
I don't think the two cases are equivalent. A `String` without any annotations is not different than a `String` without any annotations. The two instances will be `equals()` but they'll be of different types. That’s weird.